File: dayTwo/java/BookSuggestionSystem.java
import java.util.ArrayList;
import java.util.List;


public class BookSuggestionSystem {
    List<String> myBooks = new ArrayList<>();
    int bookCount = 0;

    public int getBookCount() {
        return bookCount;
    }

    public String suggestBook() {
        int bookIndex = (int) (Math.random() * bookCount);
        int pageNumber = (int) (Math.random() * 100) + 1;

        String recommendedBook = myBooks.get(bookIndex);

        return String.format("Book Title: %s\nPage: %d", recommendedBook, pageNumber);
    }

    public void addBook(String bookTitle) {
        if (!myBooks.contains(bookTitle)) {
            myBooks.add(bookTitle);
            bookCount++;
        }
        else
            throw new IllegalArgumentException("Book already exists!!!");
   }

    public void removeBook(String bookTitle) {
        boolean result = myBooks.remove(bookTitle);

        if (result)
            bookCount--;
        else
            throw new IllegalArgumentException("Book does not exist!!!");
    }

    public void updateBook(String oldBookTitle, String newBookTitle) {
        int index = myBooks.indexOf(oldBookTitle);

        if (index == -1)
            throw new IllegalArgumentException("Book does not exist!!!");

        myBooks.set(index, newBookTitle);
    }

    public List<String> getBookList() {
        return List.copyOf(myBooks);
    }
}

File: dayTwo/python/book_suggestion_system.py
import random

class BookSuggestionSystem:

    def __init__(self):
        self.my_books = []

    def get_book_count(self):
        return len(self.my_books)

    def suggest_book(self):
        book_count = len(self.my_books)
        if book_count == 0:
            raise ValueError("No books available!!!")

        book_index = random.randint(0, book_count - 1)
        page_number = random.randint(1, 100)

        recommended_book = self.my_books[book_index]

        return f"Book Title: {recommended_book}\nPage: {page_number}"

    def add_book(self, book_title):
        if book_title not in self.my_books:
            self.my_books.append(book_title)
        else:
            raise ValueError("Book already exists!!!")

    def remove_book(self, book_title):
        if book_title in self.my_books:
            self.my_books.remove(book_title)
        else:
            raise ValueError("Book does not exist!!!")

    def update_book(self, old_book_title, new_book_title):
        if old_book_title not in self.my_books:
            raise ValueError("Book does not exist!!!")

        index = self.my_books.index(old_book_title)
        self.my_books[index] = new_book_title

    def get_book_list(self):
        return self.my_books.copy()
